kore/src/Kore/Step/Simplification/TermLike.hs

```diff
@@ -153,8 +160,10 @@ simplifyInternal term predicate = simplifyInternalWorker term
     simplifyChildren = traverse simplifyInternalWorker
 
     simplifyInternalWorker termLike =
-        tracer termLike $
-        let doNotSimplify = return (OrPattern.fromTermLike termLike)
+        assertSimplifiedResults $ tracer termLike $
```

Comment:
Shouldn't this start with checking whether `termLike` is already simplified?

Reply:
I want to do that in another pull request, after I am satisfied that the simplifier is _actually_ idempotent (probably adding property tests).